### DISCUSSION TITLE
MSI Package

### DISCUSSION THREAD

**Discussion by alec-hs** [Kong/insomnia#3658] [Ideas]
**Is your feature request related to a problem? Please describe.**
Deploying to multiple users via tools like MS Intune/ PDQ Deploy is much easier when devs provide MSI installers. 

**Describe the solution you'd like**
An MSI package with silent/user-based/whole-machine switches.

**Comment by dimitropoulos** [comment]
Do you happen to have any pointer to another electron app that does this.  E.g. I didn't see vscode offering this.  It would be helpful to gauge how much work it might be.

**Reply by Mojo-OG** [reply]
https://code.visualstudio.com/download

They have offered for a long time a separate 'System' installer that installs under the machine Program Files/(x86) directory... Can you not simply do the same?

**Comment by alec-hs** [comment]
This npm package makes most of the work trivial if you already use WIX or most other windows CI systems. 

https://www.npmjs.com/package/electron-wix-msi

Here's a few Open Source examples that use it.

https://github.com/tiagovtristao/clipmir-desktop
https://github.com/Pexeus/LCSS
https://github.com/iPzard/media-file-renamer

**Reply by OraDotNetDev** [reply]
There is a _"Machine-wide Install"_ [option within Squirrel](https://github.com/Squirrel/Squirrel.Windows/blob/develop/docs/using/machine-wide-installs.md), I think this could go part way towards providing an "Enterprise" installation mechanism.

However I don't think that this quite meets Enterprise Management "_best practice_".

- Software is installed at Machine level under "Program Files"
- Users are not be able to modify applications files.
 - A single verified good payload is used to deploy installations and control upgrades (at Machine level). 
 
For more detail around this see this [Microsoft Teams Feedback request](https://feedbackportal.microsoft.com/feedback/idea/96fe0515-272e-ec11-b6e6-00224827bbc2). Teams uses a very similar (same?) mechanism as Squirrel's machine-wide installs.
